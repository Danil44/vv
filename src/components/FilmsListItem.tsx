import React from 'react';
import { jsx, css } from '@emotion/core';
import { mainOrange } from '../colors';
/**@jsx jsx */

type FilmsListItemProps = {
  imageSrc?: string;
  name: string;
  active?: boolean;
};

export default function FilmsListItem({
  imageSrc,
  name,
  active = false,
}: FilmsListItemProps) {
  return (
    <li
      css={css`
        width: calc(20% - 12px);
        display: inline-block;
        min-height: 120px;
        max-height: calc(20% * 5);
        margin-bottom: 10px;
        border: 2px solid ${active ? mainOrange : 'transparent'};
        transform: ${active ? 'scale(1.03)' : 'scale(1)'};

        &:not(:nth-of-type(5n + 5)) {
          margin-right: 10px;
        }
      `}
    >
      {imageSrc && (
        <img
          src={imageSrc}
          css={css`
            width: 100%;
            height: 100%;
            object-fit: cover;
          `}
          alt={`${name}'s poster`}
        />
      )}
    </li>
  );
}
