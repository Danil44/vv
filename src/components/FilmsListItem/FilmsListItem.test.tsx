import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import FilmsListItem from './FilmsListItem';

describe('<FilmsListItem />', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<FilmsListItem name="" />);
  });

  it('has not changed', () => expect(wrapper.html()).toMatchSnapshot());

  it('should render image with provided name props in the alt attribute if the prop imageSrc is present', () => {
    wrapper.setProps({ imageSrc: 'src', name: 'Film' });

    expect(wrapper.find('img').exists()).toEqual(true);
    expect(wrapper.find('img').props().alt).toBe("Film's poster");
  });
});
