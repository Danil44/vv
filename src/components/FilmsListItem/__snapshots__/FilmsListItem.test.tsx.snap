// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmsListItem /> has not changed 1`] = `"<li class=\\"css-1m28q18-FilmsListItem\\"></li>"`;
